### PR TITLE
ICMSLST-665 Convert multifile to single file upload for FIR views.

### PR DESCRIPTION
**Pull request by MattHolmes123**
Changes:

- Update FIR templates to add single files
- Add FIR views to redirect to correct template
- Add check_application_permission utility function

### CODE REVIEW

**Comment by oskusalerma** [comment]
Spelling on commit messag:

- tempates -> templates
- refirect -> redirect

File: web/domains/case/views.py [hunk 1]
```diff
@@ -794,48 +818,162 @@ def close_fir(
...
 
 
+@login_required
+@permission_required("web.reference_data_access", raise_exception=True)
+def add_fir_file(
+    request: HttpRequest, *, application_pk: int, fir_pk: int, case_type: str
+) -> HttpResponse:
+    redirect_url = "case:edit-fir"
+    base_template = "web/domains/case/manage/base.html"
+    return _add_fir_file(request, application_pk, fir_pk, case_type, redirect_url, base_template)
```
Comment (oskusalerma) [inline web/domains/case/views.py]: empty line before returns please

File: web/domains/case/views.py [hunk 2]
```diff
@@ -794,48 +818,162 @@ def close_fir(
...
+    with transaction.atomic():
+        model_class = _get_class_imp_or_exp_or_access(case_type)
+        application: ImpOrExpOrAccess = get_object_or_404(model_class, pk=application_pk)
+        check_application_permission(application, request.user, case_type)
+
+        fir = get_object_or_404(application.further_information_requests, pk=fir_pk)
+
+        if request.POST:
+            form = forms.DocumentForm(data=request.POST, files=request.FILES)
+            if form.is_valid():
```
Comment (oskusalerma) [inline web/domains/case/views.py:857]: empty line before

File: web/domains/case/views.py [hunk 3]
```diff
@@ -794,48 +818,162 @@ def close_fir(
...
 
 
+@login_required
+@permission_required("web.reference_data_access", raise_exception=True)
+def add_fir_file(
+    request: HttpRequest, *, application_pk: int, fir_pk: int, case_type: str
+) -> HttpResponse:
+    redirect_url = "case:edit-fir"
+    template_name = "web/domains/case/fir/add-fir-file.html"
+    return _add_fir_file(request, application_pk, fir_pk, case_type, redirect_url, template_name)
```
Comment (oskusalerma) [inline web/domains/case/views.py:828]: empty line before all returns. i'm not gonna comment on each one.

File: web/domains/case/views.py [hunk 4]
```diff
@@ -794,48 +818,162 @@ def close_fir(
...
+
+@login_required
+@require_GET
+def view_fir_file(
+    request: HttpRequest,
+    *,
+    application_pk: int,
+    fir_pk: int,
+    file_pk: int,
+    case_type: str,
```
Comment (oskusalerma) [inline web/domains/case/views.py:897]: this fits on one line when you take out the trailing comma

File: web/templates/web/domains/case/fir/add-fir-file.html [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{% extends "web/domains/case/manage/base.html" %}
+{% import "forms/forms.html" as forms %}
+{% import "forms/fields.html" as fields %}
+
+{% block content_actions %}
+  <div class="content-actions">
+    <ul class="menu-out flow-across">
+      <li>
+        <a
+          href="{{ icms_url(redirect_url, kwargs={'application_pk': process.pk, 'fir_pk': fir.pk, 'case_type': case_type}) }}"
```
Comment (oskusalerma) [inline web/templates/web/domains/case/fir/add-fir-file.html:10]: isn't `prev_link` a more accurate name for this template variable? nothing to do with redirect_url

**Review by oskusalerma — CHANGES_REQUESTED** [review]

File: web/templates/web/domains/case/fir/add-fir-file.html (hunk 5 above)
Comment (MattHolmes123) [reply web/templates/web/domains/case/fir/add-fir-file.html:10]: Yes I suppose so (it was just the view name for it)

**Review by oskusalerma — APPROVED** [review]

**Pull request merged**